src/pages/index.js

```diff
@@ -145,54 +144,135 @@ function openModal(modal) {
 }
 
 function closeModal(modal) {
-  modal.classList.remove("modal_opened");
+  modal.classList.remove(modalLikedCSSClass);
 
   modal.removeEventListener("click", handleModalOutsideClick);
   modal.removeEventListener("keydown", handleModalEscapeKey);
 }
 
+function addCardToCardList(card, method = "prepend") {
+  const cardElement = getCardElement(card);
+  cardsList[method](cardElement);
+}
+
+function handleModalOutsideClick(evt) {
+  if (evt.target.classList.contains(modalLikedCSSClass)) {
+    closeModal(evt.target);
+  }
+}
+
+function handleModalEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    closeModal(evt.currentTarget);
+  }
+}
+
+function handleToggleLikeCard(evt, data) {
+  api
+    .changeCardLikeStatus({ id: data._id, isLiked: data.isLiked })
+    .then(() => {
+      evt.target.classList.toggle(cardLikedCSSClass);
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+
+  openModal(deleteCardModal);
+}
+
+//Form submit handlers
 function handleEditProfileSubmit(evt) {
   evt.preventDefault();
 
-  profileNameLabel.textContent = editProfileModalNameInput.value;
-  profileDescriptionParagraph.textContent =
-    editProfileModalDescriptionInput.value;
+  editProfileSubmitButton.textContent = savingSaveButtonText;
 
   disableButton(editProfileSubmitButton, validationConfig);
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Reply:
My concern was a user spamming the button in the case where the promise takes a long time to resolve. This would cause redundant fetch's.
What about disabling it like I was doing, and reenabling it in the catch block?

Reply:
> What about disabling it like I was doing, and reenabling it in the catch block?

Right. You've done everything correct in `handleSubmit`